src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+        }
+
+        String tokenEndpointAuthMethod = (String) oAuthApplicationInfo.getParameter(
+                OktaConstants.CLIENT_TOKEN_ENDPOINT_AUTH_METHOD);
+        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+            paramMap.put(OktaConstants.CLIENT_TOKEN_ENDPOINT_AUTH_METHOD, tokenEndpointAuthMethod);
+        } else {
+            handleException("Mandatory parameter token_endpoint_auth_method is missing");
+        }
+
+        String clientUri = (String) oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_URI);
+        if (StringUtils.isNotEmpty(clientUri)) {
+            paramMap.put(OktaConstants.CLIENT_URI, clientUri);
+        }
+
+        String logoUri = (String) oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_LOGO_URI);
+        if (StringUtils.isNotEmpty(logoUri)) {
+            paramMap.put(OktaConstants.CLIENT_LOGO_URI, logoUri);
+        }
+
+        String initiateLoginUri = (String) oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_INITIATE_LOGIN_URI);
+        if (StringUtils.isNotEmpty(initiateLoginUri)) {
+            paramMap.put(OktaConstants.CLIENT_INITIATE_LOGIN_URI, initiateLoginUri);
+        }
+
+        String applicationType = (String) oAuthApplicationInfo.getParameter(OktaConstants.CLIENT_APPLICATION_TYPE);
+        if (StringUtils.isNotEmpty(applicationType)) {
+            paramMap.put(OktaConstants.CLIENT_APPLICATION_TYPE, applicationType);
+        } else {
+            handleException("Mandatory parameter application_type is missing");
+        }
+
+        return JSONObject.toJSONString(paramMap);
+    }
+
+    /**
+     * This method will create {@code OAuthApplicationInfo} object from a Map of Attributes.
+     *
+     * @param responseMap Response returned from server as a Map
+     * @return OAuthApplicationInfo object will return.
+     */
+    private OAuthApplicationInfo createOAuthAppInfoFromResponse(Map responseMap) {
+        OAuthApplicationInfo appInfo = new OAuthApplicationInfo();
+        String clientName = (String) responseMap.get(OktaConstants.CLIENT_NAME);
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Create OAuth app info from response for the application: %s",
+                    clientName));
+        }
+        appInfo.setClientName(clientName);
+        appInfo.setClientId((String) responseMap.get(OktaConstants.CLIENT_ID));
+        appInfo.setClientSecret((String) responseMap.get(OktaConstants.CLIENT_SECRET));
+        JSONArray callbackUrl = (JSONArray) responseMap.get(OktaConstants.CLIENT_REDIRECT_URIS);
+        if (callbackUrl != null) {
+            appInfo.setCallBackURL((String) callbackUrl.toArray()[0]);
+        }
+        Object clientIdIssuedAt = responseMap.get(OktaConstants.CLIENT_ID_ISSUED_AT);
+        if (clientIdIssuedAt != null) {
+            appInfo.addParameter(OktaConstants.CLIENT_ID_ISSUED_AT, clientIdIssuedAt);
```

Comment:
Why do we need to check null here. Even if it is null, if we add a parameter will that be a problem?